Core/PrimaryGenerator.cpp

```diff
@@ -61,63 +61,18 @@ G4PrimaryVertex* PrimaryGenerator::GenerateThreeGammaVertex(
   Double_t weight;
   Double_t weight_max = event.GetWtMax() * pow(10, -1);
   Double_t rwt;
-  Double_t M_max = 7.65928 * pow(10, -6);
-  do 
+  Double_t M_max;
+  if(channel==MaterialExtension::DecayChannel::Ortho3G or channel == MaterialExtension::DecayChannel::Direct)
   {
-    weight = event.Generate();
-    weight = weight * calculate_mQED( 511., event.GetDecay(0)->E() / keV, event.GetDecay(1)->E() / keV, event.GetDecay(2)->E() / keV );
-    rwt = M_max * weight_max * (G4UniformRand());
-  } while (rwt > weight);
-
-  G4PrimaryParticle* particle[3];
-  for (int i = 0; i < 3; i++) 
+     M_max = 7.65928 * pow(10, -6);
+  } else if (channel ==MaterialExtension::DecayChannel::Para3G)
   {
-    TLorentzVector* out = event.GetDecay(i);
-    particle[i] = new G4PrimaryParticle( particleDefinition, out->Px(), out->Py(), out->Pz(), out->E() );
-    PrimaryParticleInformation* infoParticle = new PrimaryParticleInformation();
-    infoParticle->SetGammaMultiplicity(PrimaryParticleInformation::koPsGamma);
-    infoParticle->SetGeneratedGammaMultiplicity(PrimaryParticleInformation::koPsGamma);
-    infoParticle->SetIndex(i + 1);
-    infoParticle->SetGenMomentum(out->Px(), out->Py(), out->Pz());
-    particle[i]->SetUserInformation(infoParticle);
-    vertex->SetPrimary(particle[i]);
+	M_max = 2.00967*pow(10,25); 
```

Comment:
Just out of curiosity: should this number be so big? 10^25? Or rather negative power: 10^(-25)?